// @splunk/webpack-configs/component.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.create = create;
exports.default = void 0;

var _webpack = _interopRequireDefault(require("webpack"));

var _webpackMerge = _interopRequireDefault(require("webpack-merge"));

var _base = require("./base.config");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function create(options) {
  return (0, _webpackMerge.default)((0, _base.create)(options), {
    devtool: false,
    externals: /^[^/.][a-zA-Z\-0-9./]+$/,
    optimization: {
      minimize: false
    },
    output: {
      filename: '[name].js',
      libraryTarget: 'commonjs2'
    },
    plugins: [// This enables developers to use the `__DEV__` variable to create code paths that will only
    // execute in dev mode and will be removed in production.
    new _webpack.default.DefinePlugin({
      __DEV__: 'process.env.NODE_ENV !== "production"'
    })]
  });
}

var _default = create();

exports.default = _default;
