// axe-html-reporter/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createHtmlReport = void 0;
const mustache_1 = __importDefault(require("mustache"));
const loadTemplate_1 = require("./util/loadTemplate");
const prepareReportData_1 = require("./util/prepareReportData");
const prepareAxeRules_1 = require("./util/prepareAxeRules");
const saveHtmlReport_1 = require("./util/saveHtmlReport");
function createHtmlReport({ results, options }) {
    var _a;
    if (!results.violations) {
        throw new Error("'violations' is required for HTML accessibility report. Example: createHtmlReport({ results : { violations: Result[] } })");
    }
    try {
        const template = loadTemplate_1.loadTemplate();
        const preparedReportData = prepareReportData_1.prepareReportData({
            violations: results.violations,
            passes: results.passes,
            incomplete: results.incomplete,
            inapplicable: results.inapplicable,
        });
        const htmlContent = mustache_1.default.render(template, {
            url: results.url,
            violationsSummary: preparedReportData.violationsSummary,
            violations: preparedReportData.violationsSummaryTable,
            violationDetails: preparedReportData.violationsDetails,
            checksPassed: preparedReportData.checksPassed,
            checksIncomplete: preparedReportData.checksIncomplete,
            checksInapplicable: preparedReportData.checksInapplicable,
            hasPassed: Boolean(results.passes),
            hasIncomplete: Boolean(results.incomplete),
            hasInapplicable: Boolean(results.inapplicable),
            incompleteTotal: preparedReportData.checksIncomplete
                ? preparedReportData.checksIncomplete.length
                : 0,
            projectKey: options === null || options === void 0 ? void 0 : options.projectKey,
            customSummary: options === null || options === void 0 ? void 0 : options.customSummary,
            hasAxeRawResults: Boolean(results === null || results === void 0 ? void 0 : results.timestamp),
            rules: prepareAxeRules_1.prepareAxeRules(((_a = results === null || results === void 0 ? void 0 : results.toolOptions) === null || _a === void 0 ? void 0 : _a.rules) || {}),
        });
        if ((options === null || options === void 0 ? void 0 : options.doNotCreateReportFile) === true) {
            console.info('Report file will not be created because user passed options.doNotCreateReportFile = true. Use HTML output of the function to create report file');
        }
        else {
            saveHtmlReport_1.saveHtmlReport({
                htmlContent,
                reportFileName: options === null || options === void 0 ? void 0 : options.reportFileName,
                outputDir: options === null || options === void 0 ? void 0 : options.outputDir,
                outputDirPath: options === null || options === void 0 ? void 0 : options.outputDirPath
            });
        }
        return htmlContent;
    }
    catch (e) {
        console.warn(`HTML report was not created due to the error ${e.message}`);
        return `Failed to create HTML report due to an error ${e.message}`;
    }
}
exports.createHtmlReport = createHtmlReport;
