// @splunk/react-ui/test-runner-jest.config.js
/* eslint-disable no-undef */
const { getJestConfig } = require('@storybook/test-runner');

// if component is passed to storybook:test then test will run for only that component
const testMatch = process.env.STORYBOOK_TEST_COMPONENT
    ? [
          `**/${process.env.STORYBOOK_TEST_COMPONENT}.stories.[jt]s?(x)`,
          `**/${process.env.STORYBOOK_TEST_COMPONENT}.examples.[jt]s?(x)`,
      ]
    : ['**/*.stories.[jt]s?(x)', '**/*.examples.[jt]s?(x)'];

module.exports = {
    // The default configuration comes from @storybook/test-runner
    ...getJestConfig(),
    testMatch,
    transform: {
        '^.+\\.stories\\.[jt]sx?$': '@storybook/test-runner/playwright/transform',
        '^.+\\.examples\\.[jt]sx?$': '@storybook/test-runner/playwright/transform',
        '^.+\\.tsx?$': 'babel-jest',
    },
    snapshotResolver: '<rootDir>/.storybook/config/snapshotResolver.js',
    reporters: [
        'default',
        [
            'jest-stare',
            {
                resultDir: `./test-reports/${process.env.STORYBOOK_TEST_BROWSER}-${process.env.STORYBOOK_THEME_FAMILY}`,
                reportTitle: `Visual A11y Test Report - ${process.env.STORYBOOK_TEST_BROWSER} ${process.env.STORYBOOK_THEME_FAMILY}`,
                reportHeadline: `Visual A11y Test Report - ${process.env.STORYBOOK_TEST_BROWSER} ${process.env.STORYBOOK_THEME_FAMILY}`,
                hidePending: true,
                resultHtml: `visual-report.html`,
            },
        ],
    ],
    globals: {
        __DEV__: true,
    },
};
