// @splunk/themes/storybook-addon-splunk-themes/themes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TooltipSectionTitle = exports.GlobalStyles = void 0;

var _styledComponents = require("styled-components");

var _theming = require("@storybook/theming");

var _mixins = _interopRequireDefault(require("../mixins"));

var _variables = _interopRequireDefault(require("../variables"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    .sb-show-main {\n        ", ";\n        background-color: ", ";\n    }\n\n    .sbdocs-h2 {\n        color: ", "\n    }\n\n    .sbdocs-h3 {\n        color: ", "\n    }\n\n    #docs-root, .sbdocs {\n        background-color: ", ";\n    }\n\n    #docs-root, .sbdocs-title {\n        color: ", ";\n    }\n\n    .sbdocs, .sbdocs-preview {\n        background-color: ", ";\n    }\n\n    .os-host, .os-host-foreign {\n        color: ", ";\n        background-color: ", ";\n    }\n\n    .prismjs {\n        background-color: ", " !important;\n    }\n\n    .docblock-code-toggle {\n        color: ", " !important;\n        background-color: ", " !important;\n    }\n\n    #docs-root , .docblock-argstable-head, thead th {\n        color: ", " !important;\n        background-color: ", ";\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var GlobalStyles = (0, _styledComponents.createGlobalStyle)(_templateObject(), _mixins["default"].reset('block'), _variables["default"].backgroundColorPage, _variables["default"].contentColorActive, _variables["default"].contentColorActive, _variables["default"].backgroundColorPage, _variables["default"].contentColorActive, _variables["default"].backgroundColorPage, _variables["default"].contentColorDefault, _variables["default"].backgroundColorNavigation, _variables["default"].black, _variables["default"].contentColorDefault, _variables["default"].backgroundColorSection, _variables["default"].contentColorDefault, _variables["default"].backgroundColorPopup); // theming based on Storybook's <TooltipMessage>

exports.GlobalStyles = GlobalStyles;

var TooltipSectionTitle = _theming.styled.div(function (_ref) {
  var theme = _ref.theme;
  return {
    color: theme.textColor,
    fontWeight: theme.typography.weight.black,
    lineHeight: '18px',
    padding: '15px 15px 0 15px'
  };
});

exports.TooltipSectionTitle = TooltipSectionTitle;
