// @splunk/themes/storybook-addon-splunk-themes/ThemedDocsContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThemedDocsContainer = void 0;

var _react = _interopRequireDefault(require("react"));

var _blocks = require("@storybook/addon-docs/blocks");

var _theming = require("@storybook/theming");

var _useSplunkTheme2 = _interopRequireDefault(require("../useSplunkTheme"));

var _themes = require("./themes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var ThemedDocsContainer = function ThemedDocsContainer(_ref) {
  var children = _ref.children,
      context = _ref.context;

  var _useSplunkTheme = (0, _useSplunkTheme2["default"])(),
      isDark = _useSplunkTheme.isDark;

  var themedContext = _objectSpread(_objectSpread({}, context), {}, {
    storyById: function storyById(id) {
      var _storyContext$paramet, _storyContext$paramet2;

      var storyContext = context.storyById(id);
      return _objectSpread(_objectSpread({}, storyContext), {}, {
        parameters: _objectSpread(_objectSpread({}, storyContext === null || storyContext === void 0 ? void 0 : storyContext.parameters), {}, {
          __isArgsStory: true,
          docs: _objectSpread(_objectSpread({}, storyContext === null || storyContext === void 0 ? void 0 : (_storyContext$paramet = storyContext.parameters) === null || _storyContext$paramet === void 0 ? void 0 : _storyContext$paramet.docs), {}, {
            theme: isDark ? _theming.themes.dark : _theming.themes.light
          }),
          controls: _objectSpread(_objectSpread({}, storyContext === null || storyContext === void 0 ? void 0 : (_storyContext$paramet2 = storyContext.parameters) === null || _storyContext$paramet2 === void 0 ? void 0 : _storyContext$paramet2.controls), {}, {
            theme: isDark ? _theming.themes.dark : _theming.themes.light
          })
        })
      });
    }
  });

  return /*#__PURE__*/_react["default"].createElement(_blocks.DocsContainer, {
    context: themedContext
  }, /*#__PURE__*/_react["default"].createElement(_themes.GlobalStyles, null), children);
};

exports.ThemedDocsContainer = ThemedDocsContainer;
