// @splunk/themes/prisma/dataViz.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.highLow = exports.sequential = exports.divergent = exports.categorical = exports.staticColors = void 0;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/**
 *  ## Data viz
 *
 * Colors should be used in their prescribed order.
 * Don't mix-and-match between sets in the same visualization.
 *
 * ### Static colors
 *
 * @colorSet verbose
 */
var staticColors = {
  static1: '#7B56DB',
  static2: '#009CEB',
  static3: '#00CDAF',
  static4: '#DD9900',
  static5: '#FF677B',
  static6: '#CB2196',
  static7: '#813193',
  static8: '#0051B5',
  static9: '#008C80',
  static10: '#99B100',
  static11: '#FFA476',
  static12: '#FF6ACE',
  static13: '#AE8CFF',
  static14: '#00689D',
  static15: '#00490A',
  static16: '#465D00',
  static17: '#9D6300',
  static18: '#F6540B',
  static19: '#FF969E',
  static20: '#E47BFE'
};
/**
 * ### Categorical 1D
 *
 * @colorSet verbose
 */

exports.staticColors = staticColors;
var categorical1D = {
  categorical1D1: '#5C33FF',
  categorical1D2: '#207865',
  categorical1D3: '#AD3F20',
  categorical1D4: '#003E80',
  categorical1D5: '#78062A',
  categorical1D6: '#2F8811',
  categorical1D7: '#555555'
};
/**
 * ### Categorical 1L
 *
 * @colorSet verbose
 */

var categorical1L = {
  categorical1L1: '#9980FF',
  categorical1L2: '#45D4BA',
  categorical1L3: '#FB865C',
  categorical1L4: '#66AAF9',
  categorical1L5: '#E85B79',
  categorical1L6: '#88EE66',
  categorical1L7: '#F0B000'
};
/**
 * ### Categorical 2D
 *
 * @colorSet verbose
 */

var categorical2D = {
  categorical2D1: '#1F4D5B',
  categorical2D2: '#CC0AD6',
  categorical2D3: '#017FA2',
  categorical2D4: '#D81E5B',
  categorical2D5: '#621FFF',
  categorical2D6: '#348350',
  categorical2D7: '#555555'
};
/**
 * ### Categorical 2L
 *
 * @colorSet verbose
 */

var categorical2L = {
  categorical2L1: '#5599BE',
  categorical2L2: '#FB9DFB',
  categorical2L3: '#00BBEE',
  categorical2L4: '#EE3399',
  categorical2L5: '#9980FF',
  categorical2L6: '#5FBF7F',
  categorical2L7: '#F58B00'
};

var categorical = _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, categorical1D), categorical1L), categorical2D), categorical2L);
/**
 * ### Divergent 1D
 *
 * @colorSet verbose
 */


exports.categorical = categorical;
var divergent1D = {
  divergent1D1: '#118832',
  divergent1D2: '#1C6B2D',
  divergent1D3: '#284D27',
  divergent1D4: '#333022',
  divergent1D5: '#692A21',
  divergent1D6: '#9E2520',
  divergent1D7: '#D41F1F'
};
/**
 * ### Divergent 1L
 *
 * @colorSet verbose
 */

var divergent1L = {
  divergent1L1: '#08AE37',
  divergent1L2: '#55C169',
  divergent1L3: '#A1D59C',
  divergent1L4: '#EEE8CE',
  divergent1L5: '#F4BAA9',
  divergent1L6: '#F98C83',
  divergent1L7: '#FF5E5E'
};
/**
 * ### Divergent 2D
 *
 * @colorSet verbose
 */

var divergent2D = {
  divergent2D1: '#0070F3',
  divergent2D2: '#115BAD',
  divergent2D3: '#224468',
  divergent2D4: '#333022',
  divergent2D5: '#692A21',
  divergent2D6: '#9E2520',
  divergent2D7: '#D41F1F'
};
/**
 * ### Divergent 2L
 *
 * @colorSet verbose
 */

var divergent2L = {
  divergent2L1: '#2A99FF',
  divergent2L2: '#6BB3EE',
  divergent2L3: '#ADCCDD',
  divergent2L4: '#EEE8CE',
  divergent2L5: '#F4BAA9',
  divergent2L6: '#F98C83',
  divergent2L7: '#FF5E5E'
};
/**
 * ### Divergent 3D
 *
 * @colorSet verbose
 */

var divergent3D = {
  divergent3D1: '#299986',
  divergent3D2: '#277C52',
  divergent3D3: '#24551F',
  divergent3D4: '#333022',
  divergent3D5: '#422879',
  divergent3D6: '#602CA1',
  divergent3D7: '#8747DA'
};
/**
 * ### Divergent 3L
 *
 * @colorSet verbose
 */

var divergent3L = {
  divergent3L1: '#14846C',
  divergent3L2: '#45D4BA',
  divergent3L3: '#9ADEC4',
  divergent3L4: '#EEE8CE',
  divergent3L5: '#D7BEE4',
  divergent3L6: '#C093F9',
  divergent3L7: '#9156DD'
};
/**
 * ### Divergent 4D
 *
 * @colorSet verbose
 */

var divergent4D = {
  divergent4D1: '#0D8387',
  divergent4D2: '#1A6765',
  divergent4D3: '#264C44',
  divergent4D4: '#333022',
  divergent4D5: '#693623',
  divergent4D6: '#9F3B23',
  divergent4D7: '#D54124'
};
/**
 * ### Divergent 4L
 *
 * @colorSet verbose
 */

var divergent4L = {
  divergent4L1: '#008287',
  divergent4L2: '#2EA39B',
  divergent4L3: '#5CC3AF',
  divergent4L4: '#EEE8CE',
  divergent4L5: '#ECA14E',
  divergent4L6: '#E3723A',
  divergent4L7: '#DA4325'
};

var divergent = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, divergent1D), divergent1L), divergent2D), divergent2L), divergent3D), divergent3L), divergent4D), divergent4L);
/**
 * ### Sequential 1D
 *
 * @colorSet verbose
 */


exports.divergent = divergent;
var sequential1D = {
  sequential1D1: '#118832',
  sequential1D2: '#669922',
  sequential1D3: '#9D9F0D',
  sequential1D4: '#CBA700',
  sequential1D5: '#D97A0D',
  sequential1D6: '#D94E17',
  sequential1D7: '#D41F1F'
};
/**
 * ### Sequential 1L
 *
 * @colorSet verbose
 */

var sequential1L = {
  sequential1L1: '#088F44',
  sequential1L2: '#2EB82E',
  sequential1L3: '#C3CC33',
  sequential1L4: '#FFD442',
  sequential1L5: '#FFA857',
  sequential1L6: '#FF7149',
  sequential1L7: '#FE3A3A'
};
/**
 * ### Sequential 2D
 *
 * @colorSet verbose
 */

var sequential2D = {
  sequential2D1: '#333022',
  sequential2D2: '#3D2830',
  sequential2D3: '#562E4C',
  sequential2D4: '#6F3468',
  sequential2D5: '#873A83',
  sequential2D6: '#A0409F',
  sequential2D7: '#B846BB'
};
/**
 * ### Sequential 2L
 *
 * @colorSet verbose
 */

var sequential2L = {
  sequential2L1: '#EEE8CE',
  sequential2L2: '#E8C7CE',
  sequential2L3: '#E1A6CD',
  sequential2L4: '#DB86CD',
  sequential2L5: '#D465CD',
  sequential2L6: '#CE44CC',
  sequential2L7: '#C723CC'
};
/**
 * ### Sequential 3D
 *
 * @colorSet verbose
 */

var sequential3D = {
  sequential3D1: '#333022',
  sequential3D2: '#253223',
  sequential3D3: '#244333',
  sequential3D4: '#245442',
  sequential3D5: '#246451',
  sequential3D6: '#237561',
  sequential3D7: '#238570'
};
/**
 * ### Sequential 3L
 *
 * @colorSet verbose
 */

var sequential3L = {
  sequential3L1: '#EEE8CE',
  sequential3L2: '#B6ECD4',
  sequential3L3: '#7EEFDA',
  sequential3L4: '#45D4BA',
  sequential3L5: '#35B9A0',
  sequential3L6: '#249F86',
  sequential3L7: '#14846C'
};
/**
 * ### Sequential 4D
 *
 * @colorSet verbose
 */

var sequential4D = {
  sequential4D1: '#333022',
  sequential4D2: '#442519',
  sequential4D3: '#64271F',
  sequential4D4: '#832A24',
  sequential4D5: '#A0312E',
  sequential4D6: '#BD3737',
  sequential4D7: '#DA3B30'
};
/**
 * ### Sequential 4L
 *
 * @colorSet verbose
 */

var sequential4L = {
  sequential4L1: '#EEE8CE',
  sequential4L2: '#F5CEBF',
  sequential4L3: '#FCB4B0',
  sequential4L4: '#F99C96',
  sequential4L5: '#F6847C',
  sequential4L6: '#DF564D',
  sequential4L7: '#DD2E2E'
};
/**
 * ### Sequential 5D
 *
 * @colorSet verbose
 */

var sequential5D = {
  sequential5D1: '#2E2E55',
  sequential5D2: '#4B1773',
  sequential5D3: '#77136A',
  sequential5D4: '#A81A45',
  sequential5D5: '#D24620',
  sequential5D6: '#D97A0D',
  sequential5D7: '#CBA700'
};
/**
 * ### Sequential 5L
 *
 * @colorSet verbose
 */

var sequential5L = {
  sequential5L1: '#EEE8CE',
  sequential5L2: '#F2DD88',
  sequential5L3: '#FFC355',
  sequential5L4: '#FF9D66',
  sequential5L5: '#FF7777',
  sequential5L6: '#EE4477',
  sequential5L7: '#DD22BB'
};
/**
 * ### Sequential 6D
 *
 * @colorSet verbose
 */

var sequential6D = {
  sequential6D1: '#1C3355',
  sequential6D2: '#005580',
  sequential6D3: '#007575',
  sequential6D4: '#118832',
  sequential6D5: '#669922',
  sequential6D6: '#9D9F0D',
  sequential6D7: '#CBA700'
};
/**
 * ### Sequential 6L
 *
 * @colorSet verbose
 */

var sequential6L = {
  sequential6L1: '#EEE8CE',
  sequential6L2: '#E7E755',
  sequential6L3: '#A3E052',
  sequential6L4: '#0AD647',
  sequential6L5: '#00BBBB',
  sequential6L6: '#1182F3',
  sequential6L7: '#6666DD'
};

var sequential = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, sequential1D), sequential1L), sequential2D), sequential2L), sequential3D), sequential3L), sequential4D), sequential4L), sequential5D), sequential5L), sequential6D), sequential6L);
/**
 * ### High low 1D
 *
 * @colorSet verbose
 */


exports.sequential = sequential;
var highLow1D = {
  highLow1DHigh: '#1C6B2D',
  highLow1DLow: '#9E2520'
};
/**
 * ### High low 1L
 *
 * @colorSet verbose
 */

var highLow1L = {
  highLow1LHigh: '#55C169',
  highLow1LLow: '#F98C83'
};
/**
 * ### High low 2D
 *
 * @colorSet verbose
 */

var highLow2D = {
  highLow2DHigh: '#115BAD',
  highLow2DLow: '#9E2520'
};
/**
 * ### High low 2L
 *
 * @colorSet verbose
 */

var highLow2L = {
  highLow2LHigh: '#6BB3EE',
  highLow2LLow: '#F98C83'
};
/**
 * ### High low 3D
 *
 * @colorSet verbose
 */

var highLow3D = {
  highLow3DHigh: '#277C52',
  highLow3DLow: '#602CA1'
};
/**
 * ### High low 3L
 *
 * @colorSet verbose
 */

var highLow3L = {
  highLow3LHigh: '#45D4BA',
  highLow3LLow: '#C093F9'
};
/**
 * ### High low 4D
 *
 * @colorSet verbose
 */

var highLow4D = {
  highLow4DHigh: '#1A6765',
  highLow4DLow: '#9F3B23'
};
/**
 * ### High low 4L
 *
 * @colorSet verbose
 */

var highLow4L = {
  highLow4LHigh: '#2EA39B',
  highLow4LLow: '#E3723A'
};

var highLow = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, highLow1D), highLow1L), highLow2D), highLow2L), highLow3D), highLow3L), highLow4D), highLow4L);

exports.highLow = highLow;

var dataViz = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, staticColors), categorical), divergent), sequential), highLow);

var _default = dataViz;
exports["default"] = _default;
